src/System.Private.CoreLib/shared/System/Text/Encoding.cs

```diff
@@ -796,9 +798,10 @@ public byte[] GetBytes(string s, int index, int count)
             // Get the char array to convert
             char[] arrChar = new char[charCount];
 
-            int index;
-            for (index = 0; index < charCount; index++)
-                arrChar[index] = chars[index];
+            fixed (char* pArrChar = arrChar)
+            {
+                Buffer.Memmove((byte*)pArrChar, (byte*)chars, (uint)charCount * sizeof(char));
```

Comment:
Can we use Array.Copy here instead? I'd expect almost-but-not-quite as good performance, but more warm fuzzies from not having unsafe code.

Reply:
Ah, missed the type of `chars`. I think my next preference would be `new Span<char>(chars, charCount).CopyTo(arrChar)`

Reply:
Sounds good to me!